### PR TITLE
MAT-9976 Top level ChoiceTypes are now displayed the same way as regu…

### PR DESCRIPTION
**Pull request by RohitKandimalla**
…lar choice type

## MADiE PR

Jira Ticket: [MAT-9976](https://jira.cms.gov/browse/MAT-9976)
(Optional) Related Tickets:

### Summary

image

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/TypeEditor.tsx [hunk 1]
```diff
@@ -270,6 +270,53 @@ const TypeEditor = ({
       : null;
   }, [extensionProfileDef]);
 
+  // For top-level choice types, always render the ChoiceType selector by path
+  // so UI does not depend on whichever concrete type happens to be selected.
+  const isChoiceTypePath = _.endsWith(structureDefinition?.id, "[x]");
+  const shouldRenderChoiceType = isChoiceTypePath && label?.includes("[x]");
+
+  if (shouldRenderChoiceType) {
+    const excludedTypes = [
```
Comment (adongare) [inline src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/TypeEditor.tsx]: I see the similar list at other place as well. can it be moved at top or some other file and reused?

Comment (adongare) [reply src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/TypeEditor.tsx]: actually its already there in fhirDefinitionServiceUtilities.TYPE_CODE_NOT_USED. if it does not include all, we can add missing ones.

**Review by adongare — APPROVED** [review]
